<a href="https://colab.research.google.com/github/svtlnbbnv/1/blob/main/%D0%A0%D0%B0%D0%B1%D0%BE%D1%82%D0%B0_%D1%81_%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D0%BC%D0%B8_colab.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Работа с данными в colab



## Загрузка датасета 


In [ ]:
!gdown --id 18yofdxO2BYBmlw8wmVHEwGFiALR2F4lC

/usr/local/lib/python3.7/dist-packages/gdown/cli.py:131: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.
  category=FutureWarning,
Downloading...
From: https://drive.google.com/uc?id=18yofdxO2BYBmlw8wmVHEwGFiALR2F4lC
To: /content/housing_market_dataset.csv
100% 29.3M/29.3M [00:00<00:00, 176MB/s]


## Подгрузка данных в питон и начало работы с ними

In [ ]:
import pandas as pd

In [ ]:
dataset = pd.read_csv('housing_market_dataset.csv') 

In [ ]:
print(dataset)

                                                     url  \
0      https://move.ru/objects/moskva_ulica_baykalska...   
1      https://move.ru/objects/prodaetsya_2-komn_kvar...   
2      https://move.ru/objects/prodaetsya_2-komn_kvar...   
3      https://move.ru/objects/prodaetsya_2-komn_kvar...   
4      https://move.ru/objects/prodaetsya_3-komn_kvar...   
...                                                  ...   
12996  https://move.ru/objects/prodaetsya_1-komnatnay...   
12997  https://move.ru/objects/prodaetsya_1-komnatnay...   
12998  https://move.ru/objects/prodaetsya_2-komnatnay...   
12999  https://move.ru/objects/prodaetsya_1-komnatnay...   
13000  https://move.ru/objects/prodaetsya_1-komnatnay...   

                                               Заголовок  \
0      Продажа 1-комнатной квартиры, 31 м², Москва, у...   
1      Продам 2-комнатную квартиру, 83.2 м², Москва, ...   
2      Продаю 2-комнатную квартиру, 89.6 м², Москва, ...   
3      Продам 2-комнатную квартиру, 55.

In [ ]:
dataset.head()

url  \
0  https://move.ru/objects/moskva_ulica_baykalska...   
1  https://move.ru/objects/prodaetsya_2-komn_kvar...   
2  https://move.ru/objects/prodaetsya_2-komn_kvar...   
3  https://move.ru/objects/prodaetsya_2-komn_kvar...   
4  https://move.ru/objects/prodaetsya_3-komn_kvar...   

                                           Заголовок  \
0  Продажа 1-комнатной квартиры, 31 м², Москва, у...   
1  Продам 2-комнатную квартиру, 83.2 м², Москва, ...   
2  Продаю 2-комнатную квартиру, 89.6 м², Москва, ...   
3  Продам 2-комнатную квартиру, 55.8 м², Москва, ...   
4  Продаю 3-комнатную квартиру, 128 м², Москва, Д...   

                                         Новостройка Этап строительства  \
0                                                NaN                NaN   
1         ЖК City Park («Сити Парк»), м. Выставочная            Отделка   
2  ЖК Silver (ЖК «Резиденции кинематографистов»),...            Отделка   
3     Новостройка «Некрасовка-Парк», 5 кв-л, корп. 1               Сдан   
4                     ЖК «Дыхание», м. Тимирязевская               Сдан   

   Год постройки   Класс жилья                    МЦК  Кол-во просмотров  \
0         1967.0           NaN                    NaN                125   
1         2020.0    Элит класс      МЦК Деловой центр                202   
2         2018.0  Бизнес класс   МЦК Ботанический сад                 20   
3         2019.0  Эконом класс  МЦК Шоссе Энтузиастов                 15   
4         2017.0    Элит класс           МЦК Окружная                 42   

   Кол-во просмотров сегодня  \
0                        3.0   
1                        1.0   
2                        1.0   
3                        1.0   
4                        3.0   

                                            Описание  ... Лифт  \
0  ВП-738 СВОБОДНАЯ ПРОДАЖА. ДОМ ПОД СНОС. ЮРИДИЧ...  ...  NaN   
1  Квартира премиум класса в ЖК CITY PARK с видам...  ...   да   
2  Продается 2-комн. квартира, площадью 89.6 м2 в...  ...  NaN   
3  Продаётся 2-к квартира площадью 55.8 кв.м. на ...  ...   да   
4  Продаётся 3-к квартира площадью 128 кв.м. на 5...  ...   да   

     Ближайшая станция метро        Расстояние до метро  \
0              м. Щелковская                    1294 м.   
1             м. Выставочная                     510 м.   
2                м. Свиблово   12 мин,             1 км   
3  м. Лермонтовский проспект  41 мин,             12 км   
4           м. Тимирязевская   3 мин,             280 м   

            Ближайшая школа          Рейтинг школы  \
0         ГБОУ школа № 1352   Место в рейтинге №47   
1         ГБОУ школа № 2055  Место в рейтинге №456   
2  ГБОУ Гимназия «Свиблово»   Место в рейтинге №28   
3         ГБОУ школа № 2053  Место в рейтинге №544   
4         ГБОУ школа № 1236   Место в рейтинге №63   

                             Адрес ближайшей школы  \
0          107207,г.Москва, Щелковское ш., д. 77 А   
1  123100, г. Москва, ул. 2-я Черногрязская, дом 7   
2     129323, г. Москва, ул. Седова, д. 4, корп. 1   
3      111674, г. Москва, улица Покровская, дом 35   
4       127254, г. Москва, улица Яблочкова, дом 10   

  Расстояние до ближайшей школы          Район  \
0                         190 м            NaN   
1                         776 м    Пресненский   
2                         619 м       Свиблово   
3                         931 м            NaN   
4                         754 м  Тимирязевский   

                      Рейтинг района Номер этажа  
0                                NaN           3  
1  14 место из 58 в рейтинге районов           4  
2  15 место из 58 в рейтинге районов           2  
3                                NaN          16  
4  31 место из 58 в рейтинге районов           5  

[5 rows x 36 columns]

In [ ]:
dataset

url  \
0      https://move.ru/objects/moskva_ulica_baykalska...   
1      https://move.ru/objects/prodaetsya_2-komn_kvar...   
2      https://move.ru/objects/prodaetsya_2-komn_kvar...   
3      https://move.ru/objects/prodaetsya_2-komn_kvar...   
4      https://move.ru/objects/prodaetsya_3-komn_kvar...   
...                                                  ...   
12996  https://move.ru/objects/prodaetsya_1-komnatnay...   
12997  https://move.ru/objects/prodaetsya_1-komnatnay...   
12998  https://move.ru/objects/prodaetsya_2-komnatnay...   
12999  https://move.ru/objects/prodaetsya_1-komnatnay...   
13000  https://move.ru/objects/prodaetsya_1-komnatnay...   

                                               Заголовок  \
0      Продажа 1-комнатной квартиры, 31 м², Москва, у...   
1      Продам 2-комнатную квартиру, 83.2 м², Москва, ...   
2      Продаю 2-комнатную квартиру, 89.6 м², Москва, ...   
3      Продам 2-комнатную квартиру, 55.8 м², Москва, ...   
4      Продаю 3-комнатную квартиру, 128 м², Москва, Д...   
...                                                  ...   
12996         Продаю 1-комнатную квартиру, 40 м², Москва   
12997        Продаем 1-комнатную квартиру, 40 м², Москва   
12998         Продам 2-комнатную квартиру, 70 м², Москва   
12999         Продам 1-комнатную квартиру, 41 м², Москва   
13000         Продаю 1-комнатную квартиру, 41 м², Москва   

                                             Новостройка Этап строительства  \
0                                                    NaN                NaN   
1             ЖК City Park («Сити Парк»), м. Выставочная            Отделка   
2      ЖК Silver (ЖК «Резиденции кинематографистов»),...            Отделка   
3         Новостройка «Некрасовка-Парк», 5 кв-л, корп. 1               Сдан   
4                         ЖК «Дыхание», м. Тимирязевская               Сдан   
...                                                  ...                ...   
12996                ЖК «Белые ночи», м. Бунинская аллея            Отделка   
12997                ЖК «Белые ночи», м. Бунинская аллея            Отделка   
12998                ЖК «Белые ночи», м. Бунинская аллея            Отделка   
12999                ЖК «Белые ночи», м. Бунинская аллея            Отделка   
13000                ЖК «Белые ночи», м. Бунинская аллея            Отделка   

       Год постройки    Класс жилья                    МЦК  Кол-во просмотров  \
0             1967.0            NaN                    NaN                125   
1             2020.0     Элит класс      МЦК Деловой центр                202   
2             2018.0   Бизнес класс   МЦК Ботанический сад                 20   
3             2019.0   Эконом класс  МЦК Шоссе Энтузиастов                 15   
4             2017.0     Элит класс           МЦК Окружная                 42   
...              ...            ...                    ...                ...   
12996         2019.0  Комфорт класс           МЦК Крымская                 50   
12997         2019.0  Комфорт класс           МЦК Крымская                 48   
12998         2019.0  Комфорт класс           МЦК Крымская                 60   
12999         2019.0  Комфорт класс           МЦК Крымская                 50   
13000         2019.0  Комфорт класс           МЦК Крымская                 47   

       Кол-во просмотров сегодня  \
0                            3.0   
1                            1.0   
2                            1.0   
3                            1.0   
4                            3.0   
...                          ...   
12996                        1.0   
12997                        1.0   
12998                        1.0   
12999                        1.0   
13000                        1.0   

                                                Описание  ... Лифт  \
0      ВП-738 СВОБОДНАЯ ПРОДАЖА. ДОМ ПОД СНОС. ЮРИДИЧ...  ...  NaN   
1      Квартира премиум класса в ЖК CITY PARK с видам...  ...   да   
2      Продается 2-комн. квартира, площадью 89

## Что за переменная dataset?

In [ ]:
type(dataset)

pandas.core.frame.DataFrame

DataFrame представляет собой прямоугольную таблицу значений, каждая строка которой называется **пример**, или же **sample**, а каждая колонка - **признак** или, что то же самое, **feature**. 

Индекс примера   | Признак_1 | Признак_2 | Признак_3 | Признак_4 | Признак_5 
-----------------|-----------|-----------|-----------|-----------|-----------
 1               | 10        | 20        | "строка 1"| True      | 3.1415926    
 2               | 15        | 10        | "строка 2"| False     | 2.71      
 3               | 12        | 40        | "строка 3"| False     | -1.12345  

## Оценка размеров датасета, наличия пропусков

In [ ]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13001 entries, 0 to 13000
Data columns (total 36 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   url                            13001 non-null  object 
 1   Заголовок                      13001 non-null  object 
 2   Новостройка                    6080 non-null   object 
 3   Этап строительства             5812 non-null   object 
 4   Год постройки                  7025 non-null   float64
 5   Класс жилья                    5812 non-null   object 
 6   МЦК                            7633 non-null   object 
 7   Кол-во просмотров              13001 non-null  int64  
 8   Кол-во просмотров сегодня      7224 non-null   float64
 9   Описание                       12263 non-null  object 
 10  Информация                     13001 non-null  object 
 11  Адрес                          13001 non-null  object 
 12  Тип здания                     12897 non-null 

! Количество записей (entries) - это количество объектов:

RangeIndex: 13001 **entries**, 0 to 13000

In [ ]:
dataset['Цена']

0         5 250 000 ₽
1        22 297 600 ₽
2        15 660 826 ₽
3         5 945 490 ₽
4        27 170 560 ₽
             ...     
12996     4 196 491 ₽
12997     4 196 491 ₽
12998     5 733 626 ₽
12999     4 130 670 ₽
13000     4 147 989 ₽
Name: Цена, Length: 13001, dtype: object

 ## Шаблоны анализа

In [ ]:
# Команда для загрузки датасета в область видимости colab 
!gdown --id 18yofdxO2BYBmlw8wmVHEwGFiALR2F4lC

/usr/local/lib/python3.7/dist-packages/gdown/cli.py:131: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.
  category=FutureWarning,
Downloading...
From: https://drive.google.com/uc?id=18yofdxO2BYBmlw8wmVHEwGFiALR2F4lC
To: /content/housing_market_dataset.csv
100% 29.3M/29.3M [00:00<00:00, 247MB/s]


In [ ]:
# Шаблон подгрузки датасета в colab
import pandas as pd
dataset = pd.read_csv('housing_market_dataset.csv') 
dataset.head()

url  \
0  https://move.ru/objects/moskva_ulica_baykalska...   
1  https://move.ru/objects/prodaetsya_2-komn_kvar...   
2  https://move.ru/objects/prodaetsya_2-komn_kvar...   
3  https://move.ru/objects/prodaetsya_2-komn_kvar...   
4  https://move.ru/objects/prodaetsya_3-komn_kvar...   

                                           Заголовок  \
0  Продажа 1-комнатной квартиры, 31 м², Москва, у...   
1  Продам 2-комнатную квартиру, 83.2 м², Москва, ...   
2  Продаю 2-комнатную квартиру, 89.6 м², Москва, ...   
3  Продам 2-комнатную квартиру, 55.8 м², Москва, ...   
4  Продаю 3-комнатную квартиру, 128 м², Москва, Д...   

                                         Новостройка Этап строительства  \
0                                                NaN                NaN   
1         ЖК City Park («Сити Парк»), м. Выставочная            Отделка   
2  ЖК Silver (ЖК «Резиденции кинематографистов»),...            Отделка   
3     Новостройка «Некрасовка-Парк», 5 кв-л, корп. 1               Сдан   
4                     ЖК «Дыхание», м. Тимирязевская               Сдан   

   Год постройки   Класс жилья                    МЦК  Кол-во просмотров  \
0         1967.0           NaN                    NaN                125   
1         2020.0    Элит класс      МЦК Деловой центр                202   
2         2018.0  Бизнес класс   МЦК Ботанический сад                 20   
3         2019.0  Эконом класс  МЦК Шоссе Энтузиастов                 15   
4         2017.0    Элит класс           МЦК Окружная                 42   

   Кол-во просмотров сегодня  \
0                        3.0   
1                        1.0   
2                        1.0   
3                        1.0   
4                        3.0   

                                            Описание  ... Лифт  \
0  ВП-738 СВОБОДНАЯ ПРОДАЖА. ДОМ ПОД СНОС. ЮРИДИЧ...  ...  NaN   
1  Квартира премиум класса в ЖК CITY PARK с видам...  ...   да   
2  Продается 2-комн. квартира, площадью 89.6 м2 в...  ...  NaN   
3  Продаётся 2-к квартира площадью 55.8 кв.м. на ...  ...   да   
4  Продаётся 3-к квартира площадью 128 кв.м. на 5...  ...   да   

     Ближайшая станция метро        Расстояние до метро  \
0              м. Щелковская                    1294 м.   
1             м. Выставочная                     510 м.   
2                м. Свиблово   12 мин,             1 км   
3  м. Лермонтовский проспект  41 мин,             12 км   
4           м. Тимирязевская   3 мин,             280 м   

            Ближайшая школа          Рейтинг школы  \
0         ГБОУ школа № 1352   Место в рейтинге №47   
1         ГБОУ школа № 2055  Место в рейтинге №456   
2  ГБОУ Гимназия «Свиблово»   Место в рейтинге №28   
3         ГБОУ школа № 2053  Место в рейтинге №544   
4         ГБОУ школа № 1236   Место в рейтинге №63   

                             Адрес ближайшей школы  \
0          107207,г.Москва, Щелковское ш., д. 77 А   
1  123100, г. Москва, ул. 2-я Черногрязская, дом 7   
2     129323, г. Москва, ул. Седова, д. 4, корп. 1   
3      111674, г. Москва, улица Покровская, дом 35   
4       127254, г. Москва, улица Яблочкова, дом 10   

  Расстояние до ближайшей школы          Район  \
0                         190 м            NaN   
1                         776 м    Пресненский   
2                         619 м       Свиблово   
3                         931 м            NaN   
4                         754 м  Тимирязевский   

                      Рейтинг района Номер этажа  
0                                NaN           3  
1  14 место из 58 в рейтинге районов           4  
2  15 место из 58 в рейтинге районов           2  
3                                NaN          16  
4  31 место из 58 в рейтинге районов           5  

[5 rows x 36 columns]

In [ ]:
# Команда просмотра сводной информации по колонкам
dataset.info()

## Обработка колонки цены - приведение к числовому типу

In [ ]:
dataset['Цена']

0         5 250 000 ₽
1        22 297 600 ₽
2        15 660 826 ₽
3         5 945 490 ₽
4        27 170 560 ₽
             ...     
12996     4 196 491 ₽
12997     4 196 491 ₽
12998     5 733 626 ₽
12999     4 130 670 ₽
13000     4 147 989 ₽
Name: Цена, Length: 13001, dtype: object

In [ ]:
price = '4 000 000 ₽'

In [ ]:
splited_price = price.split()
print(splited_price)
clean_price = ''.join(splited_price[:-1])
print(clean_price)

In [ ]:
import re
clean_price = re.sub('\D', '', price)
print(clean_price)

4000000


In [ ]:
def clear_price(price):
  return re.sub('\D', '', price)

In [ ]:
clear_price(price)

'4000000'

In [ ]:
dataset['Цена'].apply(clear_price)

0         5250000
1        22297600
2        15660826
3         5945490
4        27170560
           ...   
12996     4196491
12997     4196491
12998     5733626
12999     4130670
13000     4147989
Name: Цена, Length: 13001, dtype: object

In [ ]:
dataset['Цена']

0         5 250 000 ₽
1        22 297 600 ₽
2        15 660 826 ₽
3         5 945 490 ₽
4        27 170 560 ₽
             ...     
12996     4 196 491 ₽
12997     4 196 491 ₽
12998     5 733 626 ₽
12999     4 130 670 ₽
13000     4 147 989 ₽
Name: Цена, Length: 13001, dtype: object

In [ ]:
clean_price_column = dataset['Цена'].apply(clear_price)
clean_price_column

0         5250000
1        22297600
2        15660826
3         5945490
4        27170560
           ...   
12996     4196491
12997     4196491
12998     5733626
12999     4130670
13000     4147989
Name: Цена, Length: 13001, dtype: object

In [ ]:
dataset['Очищенная цена'] = dataset['Цена'].apply(clear_price)

In [ ]:
dataset.head()

url  \
0  https://move.ru/objects/moskva_ulica_baykalska...   
1  https://move.ru/objects/prodaetsya_2-komn_kvar...   
2  https://move.ru/objects/prodaetsya_2-komn_kvar...   
3  https://move.ru/objects/prodaetsya_2-komn_kvar...   
4  https://move.ru/objects/prodaetsya_3-komn_kvar...   

                                           Заголовок  \
0  Продажа 1-комнатной квартиры, 31 м², Москва, у...   
1  Продам 2-комнатную квартиру, 83.2 м², Москва, ...   
2  Продаю 2-комнатную квартиру, 89.6 м², Москва, ...   
3  Продам 2-комнатную квартиру, 55.8 м², Москва, ...   
4  Продаю 3-комнатную квартиру, 128 м², Москва, Д...   

                                         Новостройка Этап строительства  \
0                                                NaN                NaN   
1         ЖК City Park («Сити Парк»), м. Выставочная            Отделка   
2  ЖК Silver (ЖК «Резиденции кинематографистов»),...            Отделка   
3     Новостройка «Некрасовка-Парк», 5 кв-л, корп. 1               Сдан   
4                     ЖК «Дыхание», м. Тимирязевская               Сдан   

   Год постройки   Класс жилья                    МЦК  Кол-во просмотров  \
0         1967.0           NaN                    NaN                125   
1         2020.0    Элит класс      МЦК Деловой центр                202   
2         2018.0  Бизнес класс   МЦК Ботанический сад                 20   
3         2019.0  Эконом класс  МЦК Шоссе Энтузиастов                 15   
4         2017.0    Элит класс           МЦК Окружная                 42   

   Кол-во просмотров сегодня  \
0                        3.0   
1                        1.0   
2                        1.0   
3                        1.0   
4                        3.0   

                                            Описание  ...  \
0  ВП-738 СВОБОДНАЯ ПРОДАЖА. ДОМ ПОД СНОС. ЮРИДИЧ...  ...   
1  Квартира премиум класса в ЖК CITY PARK с видам...  ...   
2  Продается 2-комн. квартира, площадью 89.6 м2 в...  ...   
3  Продаётся 2-к квартира площадью 55.8 кв.м. на ...  ...   
4  Продаётся 3-к квартира площадью 128 кв.м. на 5...  ...   

     Ближайшая станция метро        Расстояние до метро  \
0              м. Щелковская                    1294 м.   
1             м. Выставочная                     510 м.   
2                м. Свиблово   12 мин,             1 км   
3  м. Лермонтовский проспект  41 мин,             12 км   
4           м. Тимирязевская   3 мин,             280 м   

            Ближайшая школа          Рейтинг школы  \
0         ГБОУ школа № 1352   Место в рейтинге №47   
1         ГБОУ школа № 2055  Место в рейтинге №456   
2  ГБОУ Гимназия «Свиблово»   Место в рейтинге №28   
3         ГБОУ школа № 2053  Место в рейтинге №544   
4         ГБОУ школа № 1236   Место в рейтинге №63   

                             Адрес ближайшей школы  \
0          107207,г.Москва, Щелковское ш., д. 77 А   
1  123100, г. Москва, ул. 2-я Черногрязская, дом 7   
2     129323, г. Москва, ул. Седова, д. 4, корп. 1   
3      111674, г. Москва, улица Покровская, дом 35   
4       127254, г. Москва, улица Яблочкова, дом 10   

  Расстояние до ближайшей школы          Район  \
0                         190 м            NaN   
1                         776 м    Пресненский   
2                         619 м       Свиблово   
3                         931 м            NaN   
4                         754 м  Тимирязевский   

                      Рейтинг района  Номер этажа Очищенная цена  
0                                NaN            3        5250000  
1  14 место из 58 в рейтинге районов            4       22297600  
2  15 место из 58 в рейтинге районов            2       15660826  
3                                NaN           16        5945490  
4  31 место из 58 в рейтинге районов            5       27170560  

[5 rows x 37 columns]

In [ ]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13001 entries, 0 to 13000
Data columns (total 37 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   url                            13001 non-null  object 
 1   Заголовок                      13001 non-null  object 
 2   Новостройка                    6080 non-null   object 
 3   Этап строительства             5812 non-null   object 
 4   Год постройки                  7025 non-null   float64
 5   Класс жилья                    5812 non-null   object 
 6   МЦК                            7633 non-null   object 
 7   Кол-во просмотров              13001 non-null  int64  
 8   Кол-во просмотров сегодня      7224 non-null   float64
 9   Описание                       12263 non-null  object 
 10  Информация                     13001 non-null  object 
 11  Адрес                          13001 non-null  object 
 12  Тип здания                     12897 non-null 

In [ ]:
def clear_price(price):
  return int(re.sub('\D', '', price))
dataset['Очищенная цена'] = dataset['Цена'].apply(clear_price)

In [ ]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13001 entries, 0 to 13000
Data columns (total 37 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   url                            13001 non-null  object 
 1   Заголовок                      13001 non-null  object 
 2   Новостройка                    6080 non-null   object 
 3   Этап строительства             5812 non-null   object 
 4   Год постройки                  7025 non-null   float64
 5   Класс жилья                    5812 non-null   object 
 6   МЦК                            7633 non-null   object 
 7   Кол-во просмотров              13001 non-null  int64  
 8   Кол-во просмотров сегодня      7224 non-null   float64
 9   Описание                       12263 non-null  object 
 10  Информация                     13001 non-null  object 
 11  Адрес                          13001 non-null  object 
 12  Тип здания                     12897 non-null 

In [ ]:
sorted_dataset = dataset.sort_values('Очищенная цена')
sorted_dataset

url  \
10913  https://move.ru/objects/moskva_ulica_krylatski...   
11272  https://move.ru/objects/moskva_ulica_mytnaya_d...   
10938  https://move.ru/objects/moskva_profsoyuznaya_u...   
136    https://move.ru/objects/moskva_gorod_troick_ul...   
10107  https://move.ru/objects/dvuhkomnatnaya_kvartir...   
...                                                  ...   
5354   https://move.ru/objects/moskva_eropkinskiy_per...   
5353   https://move.ru/objects/moskva_bogoslovskiy_pe...   
1270   https://move.ru/objects/moskva_ul_naberejnaya_...   
10124  https://move.ru/objects/prodam_mnogokomnatnuyu...   
10099  https://move.ru/objects/roskoshnaya_kvartira_n...   

                                               Заголовок  \
10913  Продажа 2-комнатной квартиры, 136 м², Москва, ...   
11272  Продам 4-комнатную квартиру, 125 м², Москва, у...   
10938  Продаем 3-комнатную квартиру, 118 м², Москва, ...   
136    Продажа 1-комнатной студии, 40 м², Троицк, Цен...   
10107  Продам 2-комнатную квартиру, 160 м², Москва, 4...   
...                                                  ...   
5354   Продается 5-комнатная квартира, 350 м², Москва...   
5353   Продаем 3-комнатную квартиру, 200 м², Москва, ...   
1270   Продажа апартаментов, 460 м², Москва, набережн...   
10124  Продается квартира, 418.6 м², Москва, улица Ос...   
10099  Продам 5-комнатную квартиру, 270 м², Москва, Б...   

                                             Новостройка Этап строительства  \
10913                                                NaN                NaN   
11272            ЖК «Английский квартал», м. Октябрьская               Сдан   
10938                           ЖК «Галина», г. Коньково               Сдан   
136                                                  NaN                NaN   
10107                                                NaN                NaN   
...                                                  ...                ...   
5354                   ЖК «Еропкинский», м. Кропотинская               Сдан   
5353                       ЖК «Сытинский», м. Пушкинская               Сдан   
1270   ЖК «Башня Федерация» («Запад», «Восток»), м. М...               Сдан   
10124                                                NaN                NaN   
10099        ЖК «Дом в Брюсовом переулке», м. Пушкинская               Сдан   

       Год постройки   Класс жилья                МЦК  Кол-во просмотров  \
10913         1997.0           NaN                NaN               1053   
11272         2013.0  Бизнес класс                NaN                291   
10938         2007.0  Бизнес класс                NaN               4005   
136              NaN           NaN                NaN                198   
10107            NaN           NaN       МЦК Шелепиха                608   
...              ...           ...                ...                ...   
5354          2011.0    Элит класс                NaN                818   
5353          2011.0    Элит класс                NaN                874   
1270          2016.0    Элит класс    МЦК Кутузовская                  4   
10124            NaN           NaN        МЦК Лужники                132   
10099            NaN    Элит класс  МЦК Деловой центр                549   

       Кол-во просмотров сегодня  \
10913                        1.0   
11272                        2.0   
10938                        1.0   
136                          NaN   
10107                        2.0   
...                          ...   
5354                         1.0   
5353                         1.0   
1270                         NaN   
10124                        1.0   
10099                        1.0   

                                                Описание  ...  \
10913  Продается 2к-квартира с машиноместом, располож...  ...   
11272  Продаю 4-х комнатную квартиру, м. Серпуховская...  ...   
10938  ЖК Галина. Огороженная и охраняемая территория...  ...   
136    Продается 1 комнатная квартира студия в поселк...

 ## Шаблоны анализа

In [ ]:
# Шаблон применения какой-либо функции ко всем значениям колонки
import re

def clear_price(price):
  return int(re.sub('\D', '', price))
dataset['Очищенная цена'] = dataset['Цена'].apply(clear_price)

In [ ]:
# Шаблон сортировки по значению
sorted_dataset = dataset.sort_values('Очищенная цена')
sorted_dataset